src/block/utils/SignLikeRotation.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block\utils;
+
+interface SignLikeRotation{
+
+	/**
+	 * @return int between 0 and 15
+	 */
+	public function getRotation() : int;
+
+	/**
+	 * @param int $rotation between 0 and 15
+	 *
+	 * @throws \InvalidArgumentException if `$rotation` is out of hounds
+	 * @return $this
```

Comment:
If we give bounds here, we need to give it for the others

Reply:
there is just this interface that have bounds